File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/constantes/DestinoDoFgtsEnum.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.constantes;

public enum DestinoDoFgtsEnum {
    PAGAR("Pagar", "P", "Para pagamento ao reclamante"),
    DEPOSITAR("Recolher", "D", "Para dep\u00f3sito em conta vinculada");

    private String nome;
    private String valor;
    private String mensagem;

    private DestinoDoFgtsEnum(String nome, String valor, String mensagem) {
        this.nome = nome;
        this.valor = valor;
        this.mensagem = mensagem;
    }

    public String getNome() {
        return this.nome;
    }

    public String getValor() {
        return this.valor;
    }

    public String getMensagem() {
        return this.mensagem;
    }
}

File: src/lib/pjecalc/core/constantes/destino-do-fgts-enum.ts
/**
 * PJe-Calc v2.15.1 — DestinoDoFgtsEnum
 * Porte 1:1 de: br.jus.trt8.pjecalc.negocio.constantes.DestinoDoFgtsEnum
 *
 * Destino do FGTS apurado: pago direto ao reclamante (caso de rescisão
 * antecipada) ou recolhido em conta vinculada na CEF (caso comum).
 *
 * Ref Java: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/constantes/DestinoDoFgtsEnum.java
 */

export enum DestinoDoFgtsEnum {
  /** Pagar diretamente ao reclamante. */
  PAGAR = 'P',
  /** Depositar/recolher em conta vinculada do FGTS. */
  DEPOSITAR = 'D',
}

export const DestinoDoFgtsEnumNomes: Record<DestinoDoFgtsEnum, string> = {
  [DestinoDoFgtsEnum.PAGAR]: 'Pagar',
  [DestinoDoFgtsEnum.DEPOSITAR]: 'Recolher',
};

export const DestinoDoFgtsEnumMensagens: Record<DestinoDoFgtsEnum, string> = {
  [DestinoDoFgtsEnum.PAGAR]: 'Para pagamento ao reclamante',
  [DestinoDoFgtsEnum.DEPOSITAR]: 'Para depósito em conta vinculada',
};
